Iterator traits checking causes compilation errors

During compilation with gcc 6.2.1. I get:
```shell
$ make
Scanning dependencies of target tilegen
[  1%] Building CXX object Wangscape/tilegen/CMakeFiles/tilegen.dir/partition/TilePartitionerPerlin.cpp.o
In file included from /home/hryniuk/projekty/Wangscape/Wangscape/noise/ModuleManager.h:8:0,
                 from /home/hryniuk/projekty/Wangscape/Wangscape/tilegen/partition/TilePartitionerPerlin.h:6,
                 from /home/hryniuk/projekty/Wangscape/Wangscape/tilegen/partition/TilePartitionerPerlin.cpp:1:
/home/hryniuk/projekty/Wangscape/Wangscape/noise/Reseedable.h: In member function ‘noise::Reseedable noise::Reseedable::terrace(InputIt, InputIt, bool)’:
/home/hryniuk/projekty/Wangscape/Wangscape/noise/Reseedable.h:62:42: error: type/value mismatch at argument 1 in template parameter list for ‘template<class, class> struct std::is_same’
                   std::input_iterator_tag>::value,
                                          ^
/home/hryniuk/projekty/Wangscape/Wangscape/noise/Reseedable.h:62:42: note:   expected a type, got ‘typename std::iterator_traits<_Iter>::iterator_category:: input_iterator_tag’
/home/hryniuk/projekty/Wangscape/Wangscape/noise/Reseedable.h: In member function ‘noise::Reseedable noise::Reseedable::curve(InputIt, InputIt)’:
/home/hryniuk/projekty/Wangscape/Wangscape/noise/Reseedable.h:80:55: error: type/value mismatch at argument 1 in template parameter list for ‘template<class, class> struct std::is_same’
                                std::input_iterator_tag>::value,
                                                       ^
/home/hryniuk/projekty/Wangscape/Wangscape/noise/Reseedable.h:80:55: note:   expected a type, got ‘typename std::iterator_traits<_Iter>::iterator_category:: input_iterator_tag’
make[2]: *** [Wangscape/tilegen/CMakeFiles/tilegen.dir/build.make:231: Wangscape/tilegen/CMakeFiles/tilegen.dir/partition/TilePartitionerPerlin.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:578: Wangscape/tilegen/CMakeFiles/tilegen.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
```

If I see correctly that's because of redundant `::input_iterator_tag` fragments.

--- a/Wangscape/noise/Reseedable.h
+++ b/Wangscape/noise/Reseedable.h
@@ -64,8 +64,8 @@ private:
 template<typename InputIt>
 inline Reseedable Reseedable::terrace(InputIt first, InputIt last, bool inverted)
 {
-    static_assert(std::is_same<std::iterator_traits<InputIt>::iterator_category::input_iterator_tag,
-                  std::input_iterator_tag>::value,
+    static_assert(std::is_base_of<std::input_iterator_tag,
+                    typename std::iterator_traits<InputIt>::iterator_category>::value,
                   "InputIt must have input_iterator_tag");
     static_assert(std::is_same<typename std::iterator_traits<InputIt>::value_type,
                                double>::value,
@@ -82,8 +82,8 @@ inline Reseedable Reseedable::terrace(InputIt first, InputIt last, bool inverted
 template<typename InputIt>
 inline Reseedable Reseedable::curve(InputIt first, InputIt last)
 {
-    static_assert(std::is_same<std::iterator_traits<InputIt>::iterator_category::input_iterator_tag,
-                               std::input_iterator_tag>::value,
+    static_assert(std::is_base_of<std::input_iterator_tag,
+                    typename std::iterator_traits<InputIt>::iterator_category>::value,
                   "InputIt must have input_iterator_tag");
     static_assert(std::is_same<typename std::iterator_traits<InputIt>::value_type,
                                std::pair<double, double>>::value,

